aten/src/ATen/native/UpSample.cpp

```diff
@@ -24,7 +25,10 @@ TORCH_API c10::SmallVector<int64_t, 3> compute_output_size(
     TORCH_CHECK(static_cast<int64_t>(scale_factors->size()) == spatial_dimensions);
     c10::SmallVector<int64_t, 3> ret;
     for (const auto i : c10::irange(spatial_dimensions)) {
-      const double odim = static_cast<double>(input_size[i+2]) * scale_factors.value()[i];
+      const double d = round_with_scale_factor ? 0.5 : 0.0;
+      // if round_with_scale_factor=true we perform round (i.e. int(0.5 + x)) to match opencv, scipy,
+      // scikit-image output size
+      const double odim = d + static_cast<double>(input_size[i+2]) * scale_factors.value()[i];
```

Comment:
Just for my own understanding, is there a difference/benefit of this over using `std::round` when `round_with_scale_factor=True`?

Reply:
No, there is no benefit other than "more simple" code. Otherwise it would be something like
```c++
double odim = static_cast<double>(input_size[i+2]) * scale_factors.value()[i];
if (round_with_scale_factor) {
    odim = std::round(odim);
}
```